lib/models/customer_model.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+import 'package:badiup/constants.dart' as constants;
+import 'package:badiup/models/user_model.dart';
+import 'package:badiup/models/user_setting_model.dart';
+import 'package:badiup/models/address_model.dart';
+
+class Customer extends User {
+  List<Address> shippingAddresses;
+
+  Customer({
+    String email,
+    String name,
+    RoleType role,
+    UserSetting setting,
+    DateTime created,
+    this.shippingAddresses
+  }) : super(
+    email: email,
+    name: name,
+    role: role,
+    setting: setting,
+    created: created
+  );
+
+  Future<void> setShippingAddresses() async {
```

Comment:
Can you explain why we need to store the Address as a separate entity in the database? Do we expect to query on top of this often?
If we need to keep it in the db, then I think that this setting of address model should happen seamlessly within the fromSnapshot method below.